m4/slepc.m4

```diff
@@ -16,26 +16,47 @@ AC_DEFUN([CONFIGURE_SLEPC],
 
   AS_IF([test "$enableslepc" !=  no],
         [
-          dnl Test to see if SLEPC_DIR set by user.  If not set, then
-          dnl try to autodetect in a default directory
+          dnl If SLEPC_DIR is not set, we should try PETSC_DIR
+          dnl since SLPEc could live in the same place as PETSc
           AS_IF([test "x$SLEPC_DIR" = x],
-                [export SLEPC_DIR=/usr/lib/slepc])
+                [export SLEPC_DIR=$PETSC_DIR])
 
           dnl Test to see if SLEPC_DIR and petscversion were set by user or
           dnl autodetection.  If not set, then disable slepc, print a message.
           AS_IF([test "x$SLEPC_DIR" = "x" || test "x$petscversion" = "x"],
-                [
-                  enableslepc=no
-                  AC_MSG_RESULT(<<< SLEPc disabled.  Please set your "\$SLEPC_DIR" environment variable correctly to enable SLEPc. >>>)
-                ],
-                [
-                  AC_CHECK_HEADER([$SLEPC_DIR/include/slepcversion.h],
-                                  [SLEPC_INCLUDE="-I$SLEPC_DIR/include"],
+                 [
+                   dnl Test to see if SLEPC_DIR set by user.  If not set, then
+                   dnl try to autodetect in a default directory
+                   AC_CHECK_HEADER([/usr/lib/slepc/slepcversion.h],
+                                  [
+                                   export SLEPC_DIR=/usr/lib/slepc
```

Comment:
Hmm... I don't know what the effect (if any) of `export`ing variables in m4 files is, but we don't normally do this. I *don't* think this will set `SLEPC_DIR` in the user's environment, in case that's what you were attempting to do.